libcontainer/nsenter/nsexec.c

```diff
@@ -505,6 +505,12 @@ void join_namespaces(char *nslist)
 		if (setns(ns->fd, flag) < 0)
 			bail("failed to setns into %s namespace", ns->type);
 
+		/* See https://github.com/opencontainers/runc/issues/4466. */
```

Comment:
This actually isn't specific to that issue, the actual issue is that we forgot to do `setresuid` in this case. A more descriptive comment like

```suggestion
		/*
		 * If we change user namespaces, make sure we switch to root in the
		 * namespace (this matches the logic for unshare(CLONE_NEWUSER)). Lots
		 * of things can break if we aren't the right user. See
		 * <https://github.com/opencontainers/runc/issues/4466> for one
		 * example.
		 */
```

would be nice.